gdshelpers/parts/waveguide.py

```diff
@@ -407,19 +408,31 @@ def add_route_single_circle_to(self, final_coordinates, final_angle, max_bend_st
 
         radius = min([max_bend_strength, max_poss_radius]) if max_bend_strength is not None else max_poss_radius
         d = abs(radius * np.tan(diff_angle / 2))
-
-        tmp_wg = Waveguide.make_at_port(self._current_port)
-        tmp_wg.add_straight_segment(distance[0] - d)
-        tmp_wg.add_bend(-diff_angle, radius)
+        if final_width is not None:
+            if on_line_only:
+                segments = [distance[0] - d, radius * diff_angle]
+            else:
+                segments = [distance[0] - d, radius * diff_angle, distance[1] - d]
+            segment_ratio = np.cumsum(segments / sum(segments))
+            segment_widths = [list((final_width - self.current_port.width) * ratio + self.current_port.width) for ratio
```

Comment:
what exactly is the conversion to list here good for? I'd guess it could be just omitted